pkg/controller/utils/comparison/comparison.go

```diff
@@ -73,3 +74,26 @@ func StringsContains(a []string, x string) bool {
 	}
 	return false
 }
+
+// GenerateHashFromNodeAnnotation use to generate the MD5 hash from Node annotations
+func GenerateHashFromNodeAnnotation(edsNamespace, edsName string, nodeAnnotations map[string]string) string {
```

Comment:
I find the function name confusing, the name imply that we're hashing all the node annotations but in fact we're only hashing annotations with `ExtendedDaemonSetRessourceNodeAnnotationKey` prefix. Can we adapt the function name or at least describe this in the function documentation?